### DISCUSSION TITLE
Cannot Reply on Issue 163

### DISCUSSION THREAD

**Discussion by billabongbruno** [dakanji/RefindPlus#172] [General]
### RefindPlus Version

v0.14.0.AB Release

### Device Type

UEFI PC

### Problem Description

              Revisiting this, can you download and try v0.14.0.AB to see if the issue is still present?

_Originally posted by @dakanji in https://github.com/dakanji/RefindPlus/issues/163#issuecomment-1758082401_
            

Hey there, once more.
Issue #163 is marked for collaborators only and I can't reply there, so I'll do it here instead.

So, as per usual, I kept an interest in your repo and as soon as I saw the issue marked as resolved and spotted the Hotfix (x64_RefindPlus_001400-AA_HotFix-01), I gave it a shot as soon as I could and it worked perfectly.

Having tried the new "x64-RefindPlus_001400-AB" version just now, as per your request, I am happy to say that it continues to work flawlessly.

All five OSes I run (Kali, Ubuntu, Windows 11, macOS Sonoma and ChromeOS 118 Beta) are booting without any issues, as well as the functions I have set on my end (Restart, Shutdown and Reboot to Firmware).

The only thing that's different is that with this new version, the entry name under my BIOS shows up as blank (doesn't show anything), as opposed to all previous versions, which showed up as "rEFIndPlus", after I manually set it once.
Not an issue whatsoever for me, just letting you know something changed in that department.

Sorry for the late reply, I had to compare both "config.conf" files extensively to ensure nothing was input wrong on my end, as there were several text changes within the file and a simple "Compare" with Notepad++ wasn't really helping, as almost everything was highlighted.
So I ended up visually comparing both files side by side and just going for the lines that appeared in white (the uncommented ones) as opposed to the green ones (commented ones) (with my dark theme under Notepad++).

But yes, everything is working perfectly and I truly appreciate the work and effort you have put into this project.

Thank you.

Best regards,
Bruno.

### Problem Point

Other

### Affected Items

Other Item

### Debug Log

No log required.

**Comment by billabongbruno** [comment]
Feel free to close this after you read it, as it isn't an issue, it was just a workaround for me to be able to get in touch with you in order to reply.

**Comment by dakanji** [comment]
I see. Thanks.

Strange that you should not be able to reply there since it had been reopened. In any case, better to use the discussions feature instead of opening a new issue. I will convert this.

PS. Just realised that I should have unlocked the issue and not just reopen

**Comment by dakanji** [comment]
Hello,

Noticed that you mentioned booting `ChromeOS ` but seemed that was not previously possible according to an [earlier feature request you made ](https://github.com/dakanji/RefindPlus/issues/50).
Are you now able to boot it directly or is this via a third party tool that you have installed separately?

**Comment by billabongbruno** [comment]
Hey there, once again.

So, I ended up using Grub2Win and completely hiding the GRUB loader with some tweaking.
I have not tested RefindPlus further in terms of booting ChromeOS directly as daisy-chaining Grub2Win with RP is a fully working solution.

Besides, I am unsure that I would be ever be able to, seeing as ChromeOS's load method is somewhat different to a Linux Distro (at least the ones I use, e.g.: Ubuntu and Kali).

In Ubuntu and Kali I can load the Kernel and initrd directly, as I have a true path for them.

ChromeOS's loading method under GRUB is as follows, though:

```
	img_path=/chromeos.img
	img_uuid=c0fdb95f-bdc0-4328-abf3-7d08ed773f00
	search --no-floppy --set=root --file $img_path
	loopback loop $img_path
	source (loop,12)/efi/boot/settings.cfg
	if [ -z $verbose ] -o [ $verbose -eq 0 ]; then
		linux (loop,7)$kernel boot=local noresume noswap loglevel=7 options=$options chromeos_bootsplash=$chromeos_bootsplash $cmdline_params \
			cros_secure cros_debug img_uuid=$img_uuid img_path=$img_path \
			console= vt.global_cursor_default=0 brunch_bootsplash=$brunch_bootsplash quiet
	else
		linux (loop,7)$kernel boot=local noresume noswap loglevel=7 options=$options chromeos_bootsplash=$chromeos_bootsplash $cmdline_params \
			cros_secure cros_debug img_uuid=$img_uuid img_path=$img_path
	fi
	initrd (loop,7)/lib/firmware/amd-ucode.img (loop,7)/lib/firmware/intel-ucode.img (loop,7)/initramfs.img
```

**Comment by billabongbruno** [comment]
That being said, it would in fact be nice to completely eliminate the need for a third-party tool, such as Grub2Win and have only Windows and macOS (via OpenCore) chain-loading.

(Yes, I am aware that Windows can be booted with a script as opposed to chainloading an .efi file, but it's my personal preference there).

Loading the macOS Kernel directly (instead of chainloading, but obviously keeping OC in its folder to load proper configurations) would be awesome as well, but I am unsure if that's even possible under Sonoma 14.1 and I have no clue as to how I would even approach this.

**Comment by dakanji** [comment]
Thanks.

This explains things as they stand: https://sourceforge.net/p/refind/discussion/general/thread/c91b84cfd7/#acc1

**Reply by billabongbruno** [reply]
I might investigate this further on, as might be an even more seemless option:

[Boot with Grub via rEFInd any other partitions and iso's](https://forum.xerolinux.xyz/thread-110.html)

**Comment by billabongbruno** [comment]
Roger that.

Grub2Win is a good solution for those using Windows.
For those not using it, EFI Stub work just as well, but it's a little less user-friendly to set up.

As added information, the reason I use another GRUB other than the ones included with Ubuntu and Kali (each one installs one of its own to a different folder), is because using RP and my minimalist black theme, I can achieve a genuine seemless boot into every OS, with no verbose text at all, including ChromeOS (by setting the splash as blank) and removing timeouts on everything that's chainloaded (like OpenCore).

This is the reasoning for me using this setup.

Kind of eye-candy for me.
(I can share my minimalist theme, which is heavily based on [refind-black](https://github.com/anthon38/refind-black) (props to the author) with some added features, if you want to include it as optional in your builds).


Anyway, thank you for all your hard work once again.

**Comment by dakanji** [comment]
Yes, the default theme from upstream is quite ugly and should be changed if possible.
I think it might be used due to licensing issues and anyone can find alternatives with a little motivation.

No plans to distribute themes here at this time.
